packages/vue-virtual/src/index.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  elementScroll,
+  observeElementOffset,
+  observeElementRect,
+  observeWindowOffset,
+  observeWindowRect,
+  Virtualizer,
+  windowScroll,
+  type PartialKeys,
+  type VirtualizerOptions,
+} from '@tanstack/virtual-core'
+import {
+  computed,
+  nextTick,
+  shallowRef,
+  triggerRef,
+  unref,
+  watchEffect,
+  type Ref,
+} from 'vue'
+
+type MaybeRef<T> = T | Ref<T>
+
+function useVirtualizerBase<TScrollElement, TItemElement = unknown>(
+  options: MaybeRef<VirtualizerOptions<TScrollElement, TItemElement>>,
+): Ref<Virtualizer<TScrollElement, TItemElement>> {
+  const virtualizer = new Virtualizer(unref(options))
+  const state = shallowRef(virtualizer)
+
+  watchEffect((onCleanup) => {
+    const opts = unref(options)
+
+    // If those functions use refs, we need to track them.
+    opts.getScrollElement()
+    opts.estimateSize(0)
+
+    // We don't want to track state here
+    nextTick(() => {
+      virtualizer.setOptions({
+        ...opts,
+        onChange: (instance: Virtualizer<TScrollElement, TItemElement>) => {
+          // Force an update event
+          triggerRef(state)
+          opts.onChange?.(instance)
+        },
+      })
+
+      virtualizer._willUpdate()
+    })
+
+    onCleanup(virtualizer._didMount())
```

Comment:
I modified this section to use `pauseTracking` and `resetTracking` from `@vue/reactivity`, and this fixes https://github.com/TanStack/virtual/pull/390#issuecomment-1257154850

```suggestion
    pauseTracking();
    virtualizer.setOptions({
      ...opts,
      onChange: (instance: Virtualizer<TScrollElement, TItemElement>) => {
        // Force an update event
        triggerRef(state)
        opts.onChange?.(instance)
      },
    })

    virtualizer._willUpdate()

    onCleanup(virtualizer._didMount())
    resetTracking();
```

Here is the fork: https://codesandbox.io/s/vue-virtual-forked-p1z06n

Reply:
Hi, thank you for that! I did not know those functions existed. They are really convenient. 

I've added that change. It works like a charm now.